### DISCUSSION TITLE
VINDRIKTNING reset

### DISCUSSION THREAD

**Discussion by peterkowalewski** [arendst/Tasmota#13608] [Q&A] [answered]
I compiled the master branch with USE_VINDRIKTNING and now I am getting in the log every few minutes a reset.
MQT: tele/luftsensor/INFO3 = {"Info3":{"RestartReason":"Hardware Watchdog"}}
QPC: Reset.
The sensor is connected to D2 GPIO4 in WEMOS D1.
Can anyone help?
Regards
Peter

**Comment by barbudor** [comment, marked as answer]
Does the problem happens even if you do not define D2 GPIO4 as VINDRIKTNING RX pin ?

**Reply by barbudor** [reply]
"Hardware Watchdog" means that the code is stuck into a never ending loop 
There might be such a loop in the VINDRIKTNING  when receiving data. However it should not happen if the reception of serial data is correct.
Please double check your connections.
Are you connected on the right pinn on the VINDRIKTNING device side ?
Proper common ground connection ?

**Reply by sfromis** [reply]
Another plausible cause of the hardware watchdog biting might be due to issues with power, if too weak or unstable. It can happen that "chargers" do not produce stable power, even if their amp rating is sufficient.

**Reply by peterkowalewski** [reply]
I will try to use other pin. The connection is with very short wire like 10 cm and I am getting proper data from the device.
tele/luftsensor/SENSOR = {"Time":"2021-11-09T07:32:12","CCS811":{"eCO2":548,"TVOC":22},"VINDRIKTNING":{"PM2.5":10}}
I am using standard ikea usb c power supply. The wiring is like in those figures: https://blakadder.com/vindriktning-tasmota/ 
There is also another issue on this device with CCS811 sensor. Every few hours eCO2 and TVOC are getting very high about  2000. Any ideas on this?

**Reply by sfromis** [reply]
Open the windows :-D

**Reply by peterkowalewski** [reply]
Ok will do this.  :)

**Reply by blakadder** [reply]
I've checked the logs for mine and i also have 1 daily reset at random times with either:
1. Hardware Watchdog
2. Exception 28
3. Exception 29

**Reply by Jason2866** [reply]
Have you replaced the onboard 3.3V LDO of the Wemos mini? This looks like power issues.
All clone Wemos mini have to weak LDOs (150mA). Original Wemos mini have a 500mA type!

**Reply by blakadder** [reply]
I use an AMS1117 and an ESP-M3 but it could still be power issues since I hooked to the 5v on board not next to the usb connector

**Reply by peterkowalewski** [reply]
For me it was finally a power supply problem. Since I changed the power supply, this  error disappeared.  Thank for this  good LDO hint. I will add a better LDO for the additional sensor.
